server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -721,6 +838,9 @@ private synchronized IndexService createIndexService(
             indexModule.addIndexOperationListener(operationListener);
         }
         pluginsService.onIndexModule(indexModule);
+        if (FeatureFlags.isEnabled(FeatureFlags.EXTENSIONS)) {
```

Comment:
extensionsOrchestrator is `nullable`, better to check it is not `null`

Reply:
I think we should keep the FeatureFlag check, because when the feature is moved from experimental to full it will be much easier to keep track of everywhere there are changes and checks.